addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java

```diff
@@ -460,6 +523,95 @@ public void run() {
         return null;
     }
 
+    List<String> getTreePath(HttpMessage msg) {
+        try {
+            String path = msg.getRequestHeader().getURI().getPath();
+            if (path == null || path.isEmpty()) {
+                return Collections.emptyList();
+            }
+            if ("/".equals(path) || path.startsWith("/?")) {
+                return Collections.singletonList("/");
+            }
+            String uri = msg.getRequestHeader().getURI().toString().replaceAll("/?\\?.*", "");
+            List<Context> contexts = getModel().getSession().getContextsForUrl(uri);
+            for (Context c : contexts) {
+                if (!variantChecksMap.containsKey(c.getId())) {
+                    continue;
+                }
+                VariantOpenApiChecks checks = variantChecksMap.get(c.getId());
+                for (OperationModel operation : checks.pathsWithNoParams) {
+                    if (operation.getPath().equals(uri)
+                            && operation
+                                    .getRequestMethod()
+                                    .name()
+                                    .equalsIgnoreCase(msg.getRequestHeader().getMethod())) {
+                        return extractTreePathFromUri(uri);
+                    }
+                }
+                for (Map.Entry<OperationModel, Pattern> entry :
+                        checks.pathsWithParamsRegex.entrySet()) {
+                    if (entry.getValue().matcher(uri).matches()
+                            && entry.getKey()
+                                    .getRequestMethod()
+                                    .name()
+                                    .equalsIgnoreCase(msg.getRequestHeader().getMethod())) {
+                        return extractTreePathFromUri(
+                                entry.getKey()
+                                        .getPath()
+                                        .replace("{", SessionStructure.DATA_DRIVEN_NODE_PREFIX)
+                                        .replace("}", SessionStructure.DATA_DRIVEN_NODE_POSTFIX));
+                    }
+                }
+            }
+        } catch (Exception e) {
+            LOG.debug("Error getting tree path for message: {}", msg, e);
+        }
+        return null;
+    }
+
+    private static List<String> extractTreePathFromUri(String uri) {
+        try {
+            URI target = new URI(uri, false);
+            String path = target.getPath();
+            return Arrays.asList(path.split("/"));
+        } catch (Exception e) {
+            LOG.warn("Error extracting tree path from uri: {}", uri, e);
+        }
+        return null;
+    }
+
+    private void updateSitesTreeDDNs(Context context) {
```

Comment:
When opening a session that has a definition but nothing in context the nodes are not correctly shown as in context (seems to be just a UI issue).